### PR TITLE
Initial marketplace UI implementation

### LINKED ISSUE #210: Implement core marketplace view listing all assets currently for sale
Figure out how to create a basic marketplace view that shows all assets listed for sale in a fixed price sale. 

Considerations: we need to make sure that there's a means to filter for the host to specifically point out certain assets that are listed for sale so it's not aggregating literally everything.

### PR DESCRIPTION
**Pull request by mycodecrafting**
Initial marketplace UI implementation. closes #210

### CODE REVIEW

**Comment by ricklove** [comment]
Bug: After listing the first item on marketplace, and trying to list a 2nd, got "Listing succesful"

**Comment by ricklove** [comment]
Bug: Listing 2 items in the marketplace, does not display correctly:

image

**Comment by ricklove** [comment]
Bug: Layout - Screen width < ~1040 doesn't work fit: 

image

File: src/components/Marketplace/Catalog/FeaturedToken.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          <Spacer />
+          <Text fontSize="md">Current Price: <Text as="span" fontWeight="600">{props.sale?.price} ꜩ</Text></Text>
+          <MinterButton
+            size="md"
+            variant="primaryAction"
+            w="150px"
+            my={4}
+            onClick={e => {
+              e.preventDefault();
+              setLocation(`/collection/${props.address}/token/${props.id}`, { replace: true });
```
Comment (ricklove) [inline src/components/Marketplace/Catalog/FeaturedToken.tsx:39]: This should be `replace: false`, or else the back button will not go back to marketplace.

**Comment by mycodecrafting** [comment]
> Bug: Layout - Screen width < ~1040 doesn't work fit:
> 
> image

This is unrelated to the marketplace and was reported in #244 and fixed in #249 which was just merged into develop.

**Comment by ricklove** [comment]
Bug: Cancel sell -> Reject Transaction, causes the modal to get stuck:

image

**Comment by mycodecrafting** [comment]
> Bug: Cancel sell -> Reject Transaction, causes the modal to get stuck:
> 
> image

Good bug find but unrelated to the marketplace page. Opened as issue #285

**Comment by mycodecrafting** [comment]
> Bug: After listing the first item on marketplace, and trying to list a 2nd, got "Listing succesful":
> 
> (This has not been listed yet)
> image

Unrelated to this PR. Opened as issue #286

File: src/components/Marketplace/Catalog/FeaturedToken.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import { ipfsUriToGatewayUrl } from '../../../lib/util/ipfs';
+import { AspectRatio, Box, Container, Flex, Spacer, Text, Heading, Stack } from '@chakra-ui/react';
+import { MinterButton } from '../../common';
+import { TokenMedia } from '../../common/TokenMedia';
+
+interface FeaturedTokenProps extends Token {
+  network: string;
+}
+
+export default function FeaturedToken(props: FeaturedTokenProps) {
```
Comment (ricklove) [inline src/components/Marketplace/Catalog/FeaturedToken.tsx:13]: This breaks with small width:

image

File: src/components/common/TokenMedia.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      />
+    );
+  }
+
+  if (/^video\/.*/.test(obj.type)) {
+    return (
+      <video
+        loop
+        onClick={e => e.preventDefault()}
+        onMouseEnter={e => e.currentTarget.play()}
```
Comment (ricklove) [inline src/components/common/TokenMedia.tsx:61]: Nice, I like the play on hover

**Review by ricklove — APPROVED** [review]
This is a great start. Most of it works with a few minor bugs.

**Comment by ricklove** [comment]
Placing a token for sell causes 2 transactions to be approved. I don't know if it is possible to make this one transaction.

File: src/components/common/TokenMedia.tsx (hunk 3 above)
Comment (mycodecrafting) [reply src/components/common/TokenMedia.tsx:61]: hah some people don't and want the video to play all the time. I tend to prefer the play on hover as well but I might be outvoted here.

**Comment by mycodecrafting** [comment]
> Placing a token for sell causes 2 transactions to be approved. I don't know if it is possible to make this one transaction.

It is, though unrelated to this specific PR, I did mean to fix it in here. Thanks for reminding me. I will open a new issue for tracking and fix in this PR.

File: src/components/Marketplace/Catalog/FeaturedToken.tsx (hunk 2 above)
Comment (mycodecrafting) [reply src/components/Marketplace/Catalog/FeaturedToken.tsx:13]: > This breaks with small width:
> 
> image

Can you tell me what browser version and OS you are using? It is coded that the image and content stack on top of each other in small screens. Possible Chakra-ui bug here.

**Pull request merged**